frontend/src/components/Mobile/InputField.tsx

```diff
@@ -55,44 +63,38 @@ const StyledTitle = styled.div`
 `
 
 type InputFieldProps = {
-  title?: string
+  title: string
   placeholder: string
   value?: string
   setValue?: Dispatch<SetStateAction<string>> | ((input: string) => void)
   textArea?: boolean
-  hasError?: boolean
-  onChange?: () => void
+  errors: { [x: string]: any }
+  isRequired?: boolean
+  register: UseFormRegister<FieldValues>
 }
 
-export default function InputField({
-  title,
-  placeholder,
-  value,
-  setValue,
-  textArea,
-  onChange,
-  hasError,
-}: InputFieldProps) {
-  const [inputValue, setInputValue] = useState('')
+export default function InputField(props: InputFieldProps) {
+  const RedAsterisk = <RedText>*</RedText>
+  const { title, placeholder, textArea, errors, isRequired, register } = props
 
   return (
     <Container>
-      {title && <StyledTitle>{title}</StyledTitle>}
+      {title && (
+        <StyledTitle>
+          {title}
+          {isRequired && RedAsterisk}
+          {errors[title] && <RedText>This is required</RedText>}
+        </StyledTitle>
+      )}
       {textArea ? (
         <StyledTextArea
-          hasError={hasError}
           placeholder={placeholder}
-          value={value ?? inputValue}
-          onChange={onChange ?? ((e) => (setValue ? setValue(e.target.value) : setInputValue(e.target.value)))}
           rows={4}
+          error={errors[title]}
+          {...register(title, { required: isRequired })}
```

Comment:
might want to pass in additional parameters also? since the input can have other constraints like max length etc; eg this use case `<input {...register('test', { valueAsNumber: true, validate: (value) => value === 2 ) }} />`